lib/download.gi

```diff
@@ -0,0 +1,219 @@
+      args:= [ "--quiet", "-O", opt.target, url ];
+    else
+      args:= [ "--quiet", "-O", "-", url ];
+    fi;
+    if IsBound( opt.verifyCert ) and opt.verifyCert = false then
+      Add( args, "--no-check-certificate" );
+    fi;
+    code:= Process( DirectoryCurrent(), exec, InputTextNone(), outstream, args );
+    CloseStream( outstream );
+    if code <> 0 then
+      # wget may have created the target file; try to remove it
+      if IsBound( opt.target ) and IsString( opt.target ) and
+         IsExistingFile( opt.target ) and RemoveFile( opt.target ) <> true then
+        Error( "Download cannot remove unwanted file ", opt.target );
+      fi;
+      return rec( success:= false,
+                  error:= Concatenation( "Process returned ", String( code ) ) );
+    elif not ( IsBound( opt.target ) and IsString( opt.target ) ) then
+      return rec( success:= true, result:= res );
+    else
+      return rec( success:= true );
+    fi;
+  end ) );
+
+Add( Download_Methods, rec(
+  name:= "via curl",
+  isAvailable:= function()
+    local exec;
+
+    exec:= Filename( DirectoriesSystemPrograms(), "curl" );
+    return exec <> fail and IsExecutableFile( exec );
+  end,
+  download:= function( url, opt )
+    local res, outstream, exec, args, code;
+
+    res:= "";
+    outstream:= OutputTextString( res, true );
+    exec:= Filename( DirectoriesSystemPrograms(), "curl" );
+    args:= [ "--silent", "-L", "--fail" ];
+    if IsBound( opt.verifyCert ) and opt.verifyCert = false then
+      Add( args, "-k" );
+    fi;
+    Add( args, "--output" );
+    if IsBound( opt.target ) and IsString( opt.target ) then
+      Add( args, opt.target );
+    else
+      Add( args, "-" );
+    fi;
+    Add( args, url );
+    code:= Process( DirectoryCurrent(), exec, InputTextNone(), outstream, args );
+    CloseStream( outstream );
+    if code <> 0 then
+      return rec( success:= false,
+                  error:= Concatenation( "Process returned ", String( code ) ) );
+    elif not ( IsBound( opt.target ) and IsString( opt.target ) ) then
+      return rec( success:= true, result:= res );
+    else
+      return rec( success:= true );
+    fi;
+  end ) );
+
+
+#############################################################################
+##
+#M  Download( <url>[, <opt>] )
+##
+##  Try to download the file described by the string <url>,
+##  and return a record with the components 'success' ('true' or 'false'),
+##  and 'result' (a string, only if 'success' is 'true'),
+##  and 'error' (a string, only if 'success' is 'false').
+##
+InstallMethod( Download,
+    [ "IsString" ],
+    url -> Download( url, rec() ) );
+
+InstallMethod( Download,
+    [ "IsString", "IsRecord" ],
+    function( url, opt )
+    local errors, r, res;
+
+    # Set the default for 'verifyCert' if necessary.
+    if not IsBound( opt.verifyCert ) and
+       UserPreference( "utils", "DownloadVerifyCertificate" ) = false then
+      opt.verifyCert:= false;
```

Comment:
Why not like this?
```suggestion
    if not IsBound( opt.verifyCert ) then
      opt.verifyCert:= UserPreference( "utils", "DownloadVerifyCertificate" ) ;
```

Reply:
The handling of user preferences is not foolproof. If one really wants then one can set a non-boolean value. Explicitly checking for `false` is safe.